connections/muxer-sel-in-sec-handshake.md

```diff
@@ -0,0 +1,214 @@
+
+### Multiplexer negotiation over Noise
+
+The libp2p Noise Specification allows the Noise handshake messages to carry
+early data. [Noise-Early-Data] is carried in the second and third message of
+the XX handshake pattern as illustrated in the following message sequence chart.
+The second message carries early data in the form of a list of multiplexers
+supported by the responder, ordered by preference. The initiator sends its
+supported multiplexer list in the third message of the handshake process.
+
+For security reasons the early data SHALL not be processed until the Noise
+handshake is finished. After the Noise handshake process is fully done, the
+initiator and responder will both process the received early data and select the
+multiplexer to be used. They both iterate through the responder's preferred
+multiplexer list in order, and if the multiplexer is also supported by the
+initiator, that multiplexer is selected. If no mutually supported multiplexer is
+found, the multiplexer negotiation process MUST fall back to multistream
+-selection protocol.
+
+Example: Noise handshake between peers that have a mutually supported
+multiplexer.
+    Initiator supports: ["/yamux/1.0.0", "/mplex/6.7.0"]
+    Responder supports: ["/mplex/6.7.0", "/yamux/1.0.0"]
+
+    XX:
+    -> e
+    <- e, ee, s, es, ["/mplex/6.7.0", "/yamux/1.0.0"] 
+    -> s, se, ["/yamux/1.0.0", "/mplex/6.7.0"] 
+
+    After handshake is done, both parties can arrive on the same conclusion
+    and select "/mplex/6.7.0" as the multiplexer to use.
+
+Example: Noise handshake between peers that don't have mutually supported
+multiplexers.
+    Responder supports: ["/mplex/6.7.0"]
+    Initiator supports: ["yamux/1.0.0"]
+
+    XX:
+    -> e
+    <- e, ee, s, es, ["/mplex/6.7.0"]
+    -> s, se, ["yamux/1.0.0"]
+    
+    After handshaking is done, early data processing will find no mutually
+    supported multiplexer, and falls back to multistream-selection protocol.
+
+The multiplexer selection logic SHOULD run after the Noise handshake has
+finished mutual authentication of the peers to enhance security.
+
+The format of the early data is specified in [Noise-handshake-payload]
+
+In the previous versions that do not support this feature, the Noise handshake
+messages carry an empty field of muxer list in the early data extension. When a
+version that supports this feature talks to an older version which does not
+support this feature, the multiplexer selection process on both peers will
+generate an empty list and they MUST fall back to the multistream-selection
+process to ensure backward compatibility.
+
+## Security
+
+The multiplexer list carried in the TLS ALPN extension field is part of the
+ClientHello message which is not encrypted. This feature will expose the
+supported multiplexers in plain text, but this is not a weakening of security
+posture. In the future when [ECH] is ready the multiplexer info can be protected
```

Comment:
It definitely is a weakening!
I don't think we should reference ECH here, as we don't have a deployment path for that protocol. Instead, point out that muxers are not really private information, since it's trivial to obtain that information by connecting to the node.